### PR TITLE
[WNMGDS-1478] Token build tasks & Theme updates

### PR DESCRIPTION
**Pull request by forrestbaer**
## Summary

Includes updates to several token scripts, bug fixes, theme distribution and theme styles.

Adds some `reference.scss` files which will be connected as soon as sign-off for token theme colors is validated by stakeholders.  At that time the _defaultTheme.scss files will be connected as well and some cleanup/removal of files like `shadow.scss`, `colors.scss` will also be removed from the index of sass files in those dirs to make this work with the dist script.

There is some additional code in the `utilities.ts` file which represents the ability to compare two colors for euclidian distance, those are unused at the moment but will likely be put into use in the future with the addition of a token editor. I originally had thought to use them for the dist script, then Patrick made me aware that wouldn't be as useful as in a future editor, so I'd like to keep them in for future use.

## How to test

run `yarn token:dist` from the main repo to build themes and copy them to their destinations.

Latest doc sites published with these updated styles:

Core docs site is ready at: http://design-system-demo.s3-website-us-east-1.amazonaws.com/WNMGDS-1478-token-build-tasks/core/
Healthcare docs site is ready at: http://design-system-demo.s3-website-us-east-1.amazonaws.com/WNMGDS-1478-token-build-tasks/hcgov/
Medicare docs site is ready at: http://design-system-demo.s3-website-us-east-1.amazonaws.com/WNMGDS-1478-token-build-tasks/mgov/

### CODE REVIEW

File: packages/design-system/src/styles/settings/variables/_reference.scss [hunk 1]
```diff
@@ -0,0 +1,15 @@
+// these variables are reference names for backwards compatibility with
```
Comment (forrestbaer) [inline packages/design-system/src/styles/settings/variables/_reference.scss:1]: These files are not linked in the index.scss file for this pr, but will be when themes are connected.

File: packages/design-system-tokens/sketch/cmsds-token-importer/README.md [hunk 2]
```diff
@@ -1,3 +1,17 @@
...
+
+To install the plugin, simply double click on the `cmsds-token-importer.sketchplugin` file. So long as you have Sketch installed on your computer it will install the plugin for you automatically.
+
+The plugin will appear under the Plugins menu option and can be used to select a JSON data file to import. **The import will wipe out all current component styles in the current file** depending on the type. Theme files will overwrite all Color, Layer Style and Font Definition types and complete Token files will overwrite all current Color definitions.
+
+This is the intended behavior as these styles are imported as Libraries by the CMSDS's primary Sketch UI toolkits, so each file that imports these tokens should really only include this content.
+
+## building
+
+For developers, run `yarn install && yarn build` to create a new cmsds-token-importer.sketchplugin file.
```
Comment (pwolfert) [inline packages/design-system-tokens/sketch/cmsds-token-importer/README.md:17]: I would love to get a walkthrough of this process in Sketch at some point. I tried building the tokens as json and building the plugin and importing, but I wasn't sure what I was seeing.

File: packages/design-system-tokens/src/lib/utility.ts [hunk 3]
```diff
@@ -1,32 +1,105 @@
...
 /*
- * transforms a hex value to an 8 char hex value with opacity given as a number from 0 to 1
+ * converts a hex string '#F3G1AA' to an rgb value string 'rgb(142, 24, 89)'
+ * if an optional opacity value is passed the rgba value will be returned
  */
-export const hexOpacity = (hexVal: HexValue, opacity: number): HexValue => {
-  const alpha = Math.round(opacity * 255);
-  const alphaHex = (alpha + 0x10000).toString(16).substr(-2).toUpperCase();
-  return `${hexVal}${alphaHex}`;
+export const hexToRgbString = (hex: HexValue, opacity = 100): RGBValue | RGBAValue => {
```
Comment (pwolfert) [inline packages/design-system-tokens/src/lib/utility.ts:23]: Is this used anywhere?

**Review by pwolfert — APPROVED** [review]
Nice work! I'm excited to turn this on soon

File: packages/design-system-tokens/src/themes/healthcare/defaultTheme.ts [hunk 4]
```diff
@@ -1,116 +1,39 @@
...
-  'focus-link-inverse': `0 3px ${themeColors['focus-border-inverse']}`,
-};
+const shadow = {};
 
-const DefaultTheme = makeTheme({
+const DefaultTheme = makeThemeTypes({
   description,
   color: themeColors,
-  spacer,
+  spacer: {},
```
[the remark this answers is not in the document]
Comment (forrestbaer) [reply packages/design-system-tokens/src/themes/healthcare/defaultTheme.ts:36]: I want to introduce them with component updates in a feature branch, not necessary just wanted to put all the token types back purposefully in the feature branch.

File: packages/design-system-tokens/src/lib/utility.ts (hunk 3 above)
Comment (forrestbaer) [reply packages/design-system-tokens/src/lib/utility.ts:23]: Not at the moment but I can see it maybe having use if a subsystem requires rgb/rgba values down the line.. I've tested and confirmed it works correctly so just a quality of life item.

**Pull request merged**